pkg/capabilities/utils.go

```diff
@@ -161,10 +162,10 @@ func RegisterTrigger[I, O proto.Message](
 				}
 				select {
 				case response <- tr:
-				case <-ctx.Done():
+				case <-stop:
```

Comment:
Is it not sufficient to rely on the open/closed state of the source channel?  Do we need this as well?